src/internal/components/abstract-switch/index.tsx

```diff
@@ -90,27 +91,33 @@ export default function AbstractSwitch({
             'aria-labelledby': ariaLabelledByIds.length ? joinString(ariaLabelledByIds) : undefined,
             'aria-label': ariaLabel,
           })}
-          {/*An empty div to display the outline, because the native control is invisible*/}
-          <div className={styles.outline} />
-        </div>
-        <div className={clsx(styles.content, !label && !description && styles['empty-content'])}>
+          {/*
+          An empty element to display the outline, because the native control is invisible.
+          Using span, not div for HTML validity.
+          Note: There is a CSS selector in src/toggle/styles.scss that relies on a span element selector to show focus.
```

Comment:
Would it be possible to add an integration test for that to fail if the selector or layout would change?

Reply:
I added this because I was concerned that the element selector was a bit fragile. I think what we want to assert is that there is a visible focus ring when the underlying control is focused or more to point that there is relationship between the CSS, HTML, and JS (for the data-focus-visible attribute) that results in a focus ring. As such is a visual regression appropriate here?

Can we orchestrate backstopJS to programmatically focus the inputs and take the visual regression snapshot?

Reply:
We have integration test of focus-test but it is not a part of visual regression test on github. Currently on github visual regression test only on permutation pages.